share/shutter/resources/modules/Shutter/Draw/DrawingTool.pm

```diff
@@ -1320,12 +1320,19 @@ sub update_warning_text {
 
 	my $minutes = int((time - $self->{_start_time}) / 60);
 	$minutes = 1 if $minutes == 0;
+
+	my $txt = $self->{_d}->nget(
+		"If you don't save the image, changes from the last minute will be lost",
+		"If you don't save the image, changes from the last %d minutes will be lost",
+		$minutes
+	);
+
+	$txt = sprintf($txt, $minutes) if $minutes > 1;
```

Comment:
Maybe just silence the warning somehow? This looks more complicated than before.

Or maybe use some alternative to sprintf which is more friendly to plurals in gettext?

Reply:
Disagree. To suppress warnings is a worse way than directly resolve them. You may accidentally miss some important warning with such approach.

Reply:
`no warnings foo;` can be used in a local scope, then it won't affect other places in the same file where the warning is still relevant. And it won't affect other warnings, unlike `no warnings;` I just don't know what `foo` would be here, it probably can be found in the error message...

Reply:
Yes, it will work, but why should we suppress warnings ? A warning is an indication that something is going wrong.

Reply:
I don't really mind this change. I was just grumbling that it looks more complicated now

Reply:
I've been working as a perl developer for many years already and can say that this is a part of best practices.
First, we shorten the code row to 80 characters. The row 
`sprintf($self->{_d}->nget("If you don't save the image, changes from the last minute will be lost", "If you don't save the image, changes from the last %d minutes will be lost", $minutes),$minutes,)` looks really ugly.
Second, we do the code more readable and errors-sustainable. Suppressing warnings is a bad practice, at list in the perl world. It's worth to do so in really special cases, here we don't have the such one.

Reply:
as you might see, I added tests as well on changed functionality

Reply:
> I've been working as a perl developer for many years already and can say that this is a part of best practices.

Ah! Alright then